internal/kafka/internal/services/quota/ams_quota_service.go

```
@@ -388,27 +383,55 @@ func (q amsQuotaService) DeleteQuotaForBillingModel(subscriptionID string, kafka
+		// ams billing model is a cloud specific marketplace, i.e. 'marketplace-aws'
+		// if the kafka billing model supports this cloud specific marketplace billing model, return 'marketplace'
+		if kafkaBillingModel.HasSupportForAMSBillingModel(amsBillingModel) {
+			return ocm.AMSRelatedResourceBillingModelMarketplace.String(), true
+		}
+	case shared.StringEqualsIgnoreCase(amsBillingModel, ocm.AMSRelatedResourceBillingModelStandard.String()):
+		if kafkaBillingModel.HasSupportForStandard() {
+			return ocm.AMSRelatedResourceBillingModelStandard.String(), true
+		}
+	}
+
+	//logger.Logger.V(10).Infof("ams related resource billing model not found for subscription %q and kafka billing model %q", subscriptionID, kafkaBillingModel.ID)
+	return "", false
+}
+
+// IsQuotaEntitlementActive checks if an organisation has a SKU entitlement and that entitlement is still active in AMS
 func (q amsQuotaService) IsQuotaEntitlementActive(kafka *dbapi.KafkaRequest) (bool, error) {
 	kafkaBillingModel, err := q.kafkaConfig.GetBillingModelByID(kafka.InstanceType, kafka.ActualKafkaBillingModel)
 	if err != nil {
 		return false, err
 	}
 
-	amsRelatedResourceBillingModel, err := q.getAMSRelatedResourceBillingModel(kafka.SubscriptionId, kafkaBillingModel)
-	if err != nil || shared.StringEmpty(amsRelatedResourceBillingModel) {
-		return false, err
+	// Using a new `err1` otherwise I would get `nil is not nil` error here since GetSubscriptionByID returns *ServiceError
+	subscription, ok, err1 := q.GetSubscriptionByID(kafka.SubscriptionId)
+	if err1 != nil || !ok {
+		return false, err1
+	}
+	amsRelatedResourceBillingModel, ok := q.amsBillingModelToAMSBillingModelCategory(string(subscription.ClusterBillingModel()), kafkaBillingModel)
+	if !ok {
+		logger.Logger.V(10).Infof("incompatible ams billing model found (%q) for kafka billing model %q in subscription %q",
+			string(subscription.ClusterBillingModel()), kafka.SubscriptionId, kafkaBillingModel.ID)
+		return false, nil
```

If the ams billing model can't be found, i would've thought this should return an error?

---

It wasn't in the previous implementation, so I kept the same behaviour